network-config-analyzer/CanonicalIntervalSet.py

```diff
@@ -230,27 +231,96 @@ def is_subset(self, other):
             """
             return other.start <= self.start and other.end >= self.end
 
+    def find_interval_left(self, interval):
+        """
+        find from left to right the last interval which is lower than the input interval,
+        without overlapping/touching it
+        :param CanonicalIntervalSet.Interval interval: the input interval to search for
+        :return: the index of the interval in the interval set if found or -1 if not found
+        :rtype: int
+        """
+        low = 0
+        high = len(self.interval_set)
+        while low != high:
+            mid = (low + high) // 2
+            if self.interval_set[mid].end < interval.start - 1:
+                if mid == len(self.interval_set) - 1 or not (self.interval_set[mid + 1].end < interval.start - 1):
+                    return mid
+                low = mid + 1
+            else:
+                high = mid
+        if low == len(self.interval_set):
+            low -= 1
+        if not (self.interval_set[low].end < interval.start - 1):
```

Comment:
```suggestion
        if self.interval_set[low].end >= interval.start - 1:
```
also in `find_interval_right()`